triplea-game/triplea
# Error report dialog improvements, clickable links

## Overview
Various improvements to the result dialog after submitting an error
report:
- error report link is now clickable
- errors encountered will have better messaging and a clickable 'contact
support' link
- error message is updated to return server error message if there is
one, otherwise we will be grabbing error message from any exceptions we
encounter
- refactor: simplify constructor configuration
- refactor: move dialog creation responsibility from upload action class to new confirmation dialog controller.

## Functional Changes
- Clickable links on error upload confirmation window
- Better error messaging on confirmation window when something fails.

## Manual Testing Performed
- Custom launch individual windows to tweak their UI padding/settings.
- Went through various error/success use cases,  screenshots attached
- Injected an exception in the upload background task to make sure it was displayed and we did not have an infinite background task spinner.

## Screen Shots

### Error Case 1: Client -> Server Connection Error
Ran the client with no http server listening

![screenshot from 2019-01-08 12-16-02](https://user-images.githubusercontent.com/12397753/50856897-c1b1cb00-1340-11e9-8b9c-aa9f56717acf.png)

### Error Case 2: Server -> Github API Error
Sending empty data goes on to have a validation error from github API, generated on the http server and send back to the client;

![screenshot from 2019-01-08 12-15-14](https://user-images.githubusercontent.com/12397753/50856894-c1193480-1340-11e9-90e0-59e7f8757bfa.png)

### Success Case: Client -> Server -> Github API
![screenshot from 2019-01-08 12-15-32](https://user-images.githubusercontent.com/12397753/50856896-c1193480-1340-11e9-9bf1-d75e0271d465.png)
